core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -323,8 +330,15 @@ public Builder withHeaders(Map<String, String> headers) {
       return this;
     }
 
+    public Builder withObjectMapper(ObjectMapper objectMapper) {
+      this.mapper = objectMapper;
+      return this;
+    }
+
     public HTTPClient build() {
-      return new HTTPClient(uri, baseHeaders);
+      withHeader(CLIENT_VERSION_HEADER, IcebergBuild.fullVersion());
+      withHeader(CLIENT_GIT_COMMIT_SHORT_HEADER, IcebergBuild.gitCommitShortId());
```

Comment:
I think it's better to add those 2 headers by default in `build()` rather than having a method that would need an explicit call for adding them